torch/_refs/__init__.py

```diff
@@ -449,6 +449,20 @@ def _maybe_broadcast(*args, preserve_cpu_scalar_tensors=True):
         *(t.shape if isinstance(t, TensorLike) else None for t in args)
     )
 
+    def should_expand(a: ShapeType, b: ShapeType) -> bool:
+        from torch.fx.experimental.symbolic_shapes import guard_or_true
+
+        if len(a) != len(b):
+            return True
+
+        for x, y in zip(a, b):
+            if guard_or_true(x != y):
+                return True
+            # assume no braodcasting for unabcked.
+            torch._check(x == y)
```

Comment:
How could this torch._check() ever be false? I thought guard_or_true() will guarantee it through a guarded value.

Reply:
if x= u0 and y = u1 then we cant evaluate x!=y. 
in that case we defer this to a runtime assertion using the 
   torch._check(x == y)

The actual result at runtime could be True or False based on concrete values of u0 and u1 . 
make sense?

Reply:
If we can't evaluate x!=y then won't guard_or_true return true and return?

Reply:
ops i need the check inside lol

Reply:
ops i deleted your comment by mistake lol it was 
if this is True we would have returned